### ISSUE TITLE
Per tag generation doesn't generate correctly when using .json

### ISSUE THREAD

**Issue by perkinsjr** [fuma-nama/fumadocs#478]
### To Reproduce

1. Using a OpenAPI file that is in JSON format. (https://api.unkey.dev/openapi.json)
2. Use default generation which is per tag
3. A single file will be generated versus file for each tag. 

### Current vs. Expected behavior

Expected to see a file per tag when generating but saw one large file instead. 

### Provide environment information

```bash
Node.js v22.1.0

Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.0.0: Thu May 30 21:33:22 PDT 2024; root:xnu-11215.0.31.511.2~1/RELEASE_ARM64_T8122
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 22.1.0
  npm: 10.7.0
  Yarn: N/A
  pnpm: 9.0.6
Relevant Packages:
  next: 14.2.4 // Latest available version is detected (14.2.4).
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: N/A
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Integrations (e.g Contentlayer, OpenAPI, Typescript DocsGen)

**Comment by fuma-nama** [comment]
Oops it is a docs mistake, it should be file by default. You can pass “tag” directly

**Comment by fuma-nama** [comment]
Updated the docs: https://fumadocs.vercel.app/docs/ui/openapi
Closing this as fixed on 12.1.1

**Comment by perkinsjr** [comment]
Hey!

Thanks for fixing the docs, it seems per tag generation doesn't work with the tag option flagged. 

For example 

```
import * as OpenAPI from "fumadocs-openapi";

void OpenAPI.generateFiles({
  input: ["*.json"],
  output: "../content/docs/reference",
  per: "tag",
})
```

Doesn't generate anything and no errors are caught. 

The JSON is valid and has tags for example

```
"/v1/keys.getKey": {
      "get": {
        "tags": ["keys"],
        "operationId": "getKey",
        "security": [
          {
            "bearerAuth": []
          }
        ],
        "parameters": [
          {
            "schema": {
              "type": "string",
              "minLength": 1,
              "description": "The id of the key to fetch",
              "example": "key_1234"
            },
            "required": true,
            "name": "keyId",
            "in": "query"
          },
          {
            "schema": {
              "type": "boolean",
              "nullable": true,
              "description": "Decrypt and display the raw key. Only possible if the key was encrypted when generated."
            },
            "required": false,
            "name": "decrypt",
            "in": "query"
          }
        ],
        "responses": {
          "200": {
            "description": "The configuration for a single key",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Key"
                }
              }
            }
          },
          "400": {
            "description": "The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ErrBadRequest"
                }
              }
            }
          },
          "401": {
            "description": "Although the HTTP standard specifies \"unauthorized\", semantically this response means \"unauthenticated\". That is, the client must authenticate itself to get the requested response.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ErrUnauthorized"
                }
              }
            }
          },
          "403": {
            "description": "The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401 Unauthorized, the client's identity is known to the server.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ErrForbidden"
                }
              }
            }
          },
          "404": {
            "description": "The server cannot find the requested resource. In the browser, this means the URL is not recognized. In an API, this can also mean that the endpoint is valid but the resource itself does not exist. Servers may also send this response instead of 403 Forbidden to hide the existence of a resource from an unauthorized client. This response code is probably the most well known due to its frequent occurrence on the web.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ErrNotFound"
                }
              }
            }
          },
          "409": {
            "description": "This response is sent when a request conflicts with the current state of the server.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ErrConflict"
                }
              }
            }
          },
          "429": {
            "description": "The user has sent too many requests in a given amount of time (\"rate limiting\")",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ErrTooManyRequests"
                }
              }
            }
          },
          "500": {
            "description": "The server has encountered a situation it does not know how to handle.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ErrInternalServerError"
                }
              }
            }
          }
        }
      }
    },
```
